cmake/Modules/FindPARPACK.cmake

```diff
@@ -0,0 +1,93 @@
+#CMake module to find arpack-ng libraries
+#
+#
+#
+#
+#       PARPACK_INCLUDE_DIR
+#       PARPACK_LIBRARIES
+#
+#
+#
+#
+# The user defined parpack locations
+IF(PARPACK_LIBRARIES AND PARPACK_INCLUDE_DIR)
+  SET(PARPACK_FOUND TRUE)
+  MESSAGE(STATUS "Provided parpack include path ${PARPACK_INCLUDE_DIR}")
+  MESSAGE(STATUS "Provided parpack libraries ${PARPACK_LIBRARIES}")
+ELSE()
+  # PARPACK_LIBRARIES or PARPACK_INCLUDE_DIR are not defined
+  SET(PARPACK_FOUND FALSE)
+ENDIF()
+
+IF(NOT PARPACK_FOUND)
+  MESSAGE(STATUS "Finding parpack libraries")
+  # Try to find with CMake config file of upstream parpack.
+  FIND_PACKAGE(PARPACK CONFIG NAMES arpack arpackng arpack-ng parpack parpackng parpack-ng)
+  IF(PARPACK_FOUND)
+    GET_TARGET_PROPERTY(PARPACK_INCLUDE_DIR parpack INTERFACE_INCLUDE_DIRECTORIES)
+    # Most likely arpack and parpack are packed togeher (like in Arch linux)
+    # or they share the same include directory even in splitted packages (parpack-dev depends on arpack-dev)
+    # So in this point ARPACK_INCLUDE_DIR has to be defined or the information is loaded into the arpack
+    # interface
+    IF(NOT PARPACK_INCLUDE_DIR)
+      GET_TARGET_PROPERTY(PARPACK_INCLUDE_DIR arpack INTERFACE_INCLUDE_DIRECTORIES)
+    ENDIF()
+    GET_TARGET_PROPERTY(PARPACK_LIBRARIES parpack IMPORTED_LOCATION_RELEASE)
+    # Check if a debug build type was used
+    IF(NOT PARPACK_LIBRARIES)
+      GET_TARGET_PROPERTY(PARPACK_LIBRARIES parpack IMPORTED_LOCATION_DEBUG)
+    ENDIF()
+  ENDIF()
+  # There is no parpack-config script or something went wrong with the script
+  # Fall back to manual search
+  IF(NOT PARPACK_LIBRARIES OR NOT PARPACK_INCLUDE_DIR)
+    INCLUDE(FindPackageHandleStandardArgs)
+    MESSAGE(STATUS "Manual search of parpack")
+    # Try to find PARPACK header
+    SET(ARPACKINCLUDE
+      "${ARPACK_ROOT}/include"
+      "$ENV{ARPACK_ROOT}/include"
+      "${ARPACKROOT}/include"
+      "$ENV{ARPACKROOT}/include"
```

Comment:
Should these variables start with "P"? E.g., `PARPACKINCLUDE` and maybe more importantly `PARPACK_ROOT` (and the others).
That would support the use-case when, e.g., a PARPACK package is installed at a different prefix that a ARPACK package.

Maybe, keep the `ARPACK*` variables as hints *after* `PARPACK*` variables to facilitate the case when both libraries are installed at the same prefix (which is probably most common).